git_pile/pile.py

```diff
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: LGPL-2.1+
+"""
+This module provides the ``Pile`` class, responsible for representing the
+content of the pile tree.
```

Comment:
Not sure if I would call it pile branch, because the idea of the abstraction is that `Pile()` could represent any tree (a file system directory or a git tree) containing the result of `genpatches`. In the case of a directory, it wouldn't necessarily be a checkout of the pile branch.

Reply:
Force-pushed adding an explanation about the abstraction of the pile branch.